### PR TITLE
Optimize TypedArrayObject.prototype.set, indexOf, lastIndexOf

### PR DESCRIPTION
**Pull request by ksh8281**
+ Fix binary precedence bug

### CODE REVIEW

**Comment by ksh8281** [comment]
Fix #1250

File: src/builtins/BuiltinTypedArray.cpp [hunk 1]
```diff
@@ -485,6 +485,100 @@ static Value builtinTypedArrayCopyWithin(ExecutionState& state, Value thisValue,
...
+template <const bool isForward>
+static Value fastTypedArrayIndexSearch(TypedArrayObject* arr, size_t k, size_t len, const Value& value)
+{
+    auto type = arr->typedArrayType();
+    if (type == TypedArrayType::BigInt64 || type == TypedArrayType::BigUint64) {
+        if (!value.isBigInt()) {
+            return Value(-1);
+        }
+    } else {
+        if (!value.isNumber()) {
```
Comment (clover2123) [inline src/builtins/BuiltinTypedArray.cpp:497]: What about merging `else` and `if` to one `else if` statement here?
And if these conditions rarely happen, adding `UNLIKELY` would be good too.

File: src/builtins/BuiltinTypedArray.cpp [hunk 2]
```diff
@@ -485,6 +485,100 @@ static Value builtinTypedArrayCopyWithin(ExecutionState& state, Value thisValue,
...
+    };
+
+    if (!isForward && byteK >= byteLength) {
+        return Value(-1);
+    }
+
+    if (type == TypedArrayType::BigInt64 || type == TypedArrayType::BigUint64) {
+        uint64_t argv0ToUint64 = type == TypedArrayType::BigInt64 ? bitwise_cast<uint64_t>(value.asBigInt()->toInt64()) : value.asBigInt()->toUint64();
+        while (compFn(byteK, byteLength)) {
+            if (*reinterpret_cast<uint64_t*>(&buffer[byteK]) == argv0ToUint64) {
```
Comment (clover2123) [inline src/builtins/BuiltinTypedArray.cpp:530]: Is it fine to directly access array elements regardless of Endian?
What if we use BigEndian?

File: src/builtins/BuiltinTypedArray.cpp (hunk 1 above)
Comment (ksh8281) [reply src/builtins/BuiltinTypedArray.cpp:497]: Good point!

File: src/builtins/BuiltinTypedArray.cpp (hunk 2 above)
Comment (ksh8281) [reply src/builtins/BuiltinTypedArray.cpp:528]: IMO We don't need to care endianness.
because two values are comes from same machine

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**